app/questionnaire/path_finder.py

```diff
@@ -43,22 +44,57 @@ def routing_path(
         )
 
         if section:
+            block_ids_dependent_on_when_rules = (
+                self._get_block_ids_dependent_on_when_rules(section)
+            )
+            blocks = self._get_not_skipped_blocks_in_section(
+                current_location,
+                routing_path_block_ids,
+                section,
+                block_ids_dependent_on_when_rules,
+            )
+
+            if blocks:
+                routing_path_block_ids = self._build_routing_path_block_ids(
+                    blocks, current_location, block_ids_dependent_on_when_rules
+                )
+
+        return RoutingPath(routing_path_block_ids, section_id, list_item_id, list_name)
+
+    def _get_block_ids_dependent_on_when_rules(
+        self, section: ImmutableDict
+    ) -> list[str]:
+        block_ids_dependent_on_when_rules: list[str] = []
+
+        for section_id in self.schema.get_section_ids_dependent_on_when_rules(section):
+            block_ids_dependent_on_when_rules.extend(self.routing_path(section_id))
+
+        return block_ids_dependent_on_when_rules
+
+    def _get_not_skipped_blocks_in_section(
+        self,
+        location: Location,
+        routing_path_block_ids: List[str],
+        section: ImmutableDict,
+        block_ids_dependent_on_when_rule: List[str],
+    ) -> List[Mapping]:
+        # :TODO: Fix group skipping in its own section. Routing path will be empty and therefore not checked
+        if section:
+            not_skipped_blocks: List[Mapping] = []
```

Comment:
Should use standard collection here?
```suggestion
            not_skipped_blocks: list[Mapping] = []
```